agent/main.go

```diff
@@ -142,91 +143,78 @@ func NewAgentServer() *Agent { // nolint:gocyclo
 	serv := server.NewServer(agent.cli, agent.authData, opts.PrivateKey, opts.KeepAliveInterval, opts.SingleUserPassword)
 
 	tun := tunnel.NewTunnel()
-	tun.ConnHandler = func(w http.ResponseWriter, r *http.Request) {
-		hj, ok := w.(http.Hijacker)
+	tun.ConnHandler = func(c echo.Context) error {
+		hj, ok := c.Response().Writer.(http.Hijacker)
 		if !ok {
-			http.Error(w, "webserver doesn't support hijacking", http.StatusInternalServerError)
-
-			return
-		}
-
-		if _, _, err := hj.Hijack(); err != nil {
-			http.Error(w, "failed to hijack connection", http.StatusInternalServerError)
-
-			return
+			return c.String(http.StatusInternalServerError, "webserver doesn't support hijacking")
 		}
 
-		vars := mux.Vars(r)
-		conn, ok := r.Context().Value("http-conn").(net.Conn)
-		if !ok {
-			log.WithFields(log.Fields{
-				"version": AgentVersion,
-			}).Warning("Type assertion failed")
-
-			return
+		conn, _, err := hj.Hijack()
+		if err != nil {
+			return c.String(http.StatusInternalServerError, "failed to hijack connection")
 		}
 
-		serv.Sessions[vars["id"]] = conn
-		serv.HandleConn(conn)
+		vars := c.ParamValues()
+		//preciso fechar esse httpConn?
+		httpConn := c.Request().Context().Value("http-conn").(net.Conn)
+		serv.Sessions[vars[0]] = httpConn
```

Comment:
`c.Param("id")` instead of `c.ParamValues()[0]`

Reply:
Fixed instance of c.Param("id)